src/main/java/com/mixpanel/android/viewcrawler/EditState.java

```diff
@@ -119,15 +126,20 @@ private void applyChangesFromList(View rootView, List<ViewVisitor> changes) {
             final int size = changes.size();
             for (int i = 0; i < size; i++) {
                 final ViewVisitor visitor = changes.get(i);
-                final EditBinding binding = new EditBinding(rootView, visitor, mUiThreadHandler);
-                mCurrentEdits.add(binding);
+                try {
+                    final EditBinding binding = new EditBinding(rootView, visitor, mUiThreadHandler);
+                    mCurrentEdits.add(binding);
+                } catch (ViewVisitor.LayoutUpdateException e) {
+                    mEditErrorMessage.sendErrorMessage(e.getErrorInfo());
```

Comment:
In my opinion, this is the best place to catch it. My argument is that if LayoutUpdateException happens, we can get rid of the edit, so that it won't exist in the queue any more. If we do it later, we'll still have the edit in the queue but having a flag to skip visit(), which is not ideal - 1. we don't need the edit anyways; 2. we are spending extra time on useless stuff